### ISSUE TITLE
Strange error when a generic alias shadows an imported name

### ISSUE THREAD

**Issue by decorator-factory** [microsoft/pyright#2741]
## Code
```py
from typing import TypeVar, Union, Type
from foo import Maybe  # type: ignore

_T = TypeVar("_T")

Maybe = Union[_T, str]

def get_maybe(thing: Type[_T]) -> Maybe[_T]:
    ...
```
## Errors
On the `def get_maybe` line:
```
TypeVar "Type[_T]" is not subscriptable
Expected no type arguments for class "str"
```

If you remove the `from foo import Maybe` line, the error goes away.

**Comment by erictraut** [comment]
`Maybe` is a variable, not a type alias as you intended. Type aliases cannot have multiple declarations.

**Comment by erictraut** [comment]
I agree that this condition is resulting in a cascade of other confusing errors. I've improved the error handling so the cause is clearer.

**Comment by erictraut** [comment]
This is included in pyright 1.1.198, which I just published. It will also be included in the next release of pylance.
